babycare_product_golive_checks/models/product.py

```diff
@@ -0,0 +1,264 @@
+                      " updated because (some) references are"
+                      " too long:\n%s") % (
+                          '\n'.join([i for i in error_len_ids]))
+        if error_ids:
+            text += _("\n\nFollowing product(s) is/are not updated because"
+                      " they are not config products:\n%s") % (
+                          '\n'.join([i for i in error_ids]))
+        partial = self.env['message.wizard'].create({
+            'text': text
+        })
+        return {
+            'name': _("Message"),
+            'view_mode': 'form',
+            'view_id': False,
+            'view_type': 'form',
+            'res_model': 'message.wizard',
+            'res_id': partial.id,
+            'type': 'ir.actions.act_window',
+            'nodestroy': True,
+            'target': 'new',
+            'domain': '[]',
+        }
+
+    @api.model
+    def update_ean13_images_config_product(self):
+        text = ''
+        succes_ids = []
+        error_ids = []
+        for template_id in self._context.get('active_ids'):
+            obj = self.env['product.template'].browse(template_id)
+            template_name = obj.name
+            boms = obj.bom_ids
+            if obj.type == 'consu' and len(obj.attribute_line_ids) == 1:
+                for b in boms:
+                    for variant in (b.product_id and
+                                    [b.product_id] or
+                                    b.product_tmpl_id.product_variant_ids):
+                        for l in b.bom_line_ids:
+                            if set([al.id for al in l.attribute_value_ids]).\
+                                    issubset(set([vl.id for vl
+                                             in variant.attribute_value_ids])):
+                                    prod_ean13 = l.product_id.ean13
+                                    prod_image = l.product_id.image
+                        variant.write({'ean13': prod_ean13})
+                        variant.write({'image': prod_image})
+                succes_ids.append(template_name)
+            else:
+                error_ids.append(template_name)
+        if succes_ids:
+            text += _("\n\nFollowing config product(s) is/are updated with"
+                      " EAN13 and images:\n%s") % (
+                          '\n'.join([i for i in succes_ids]))
+        if error_ids:
+            text += _("\n\nFollowing config product(s) is/are not updated"
+                      " because it is not a config product or a config"
+                      " product with more than two attributes:\n%s") % (
+                          '\n'.join([i for i in error_ids]))
+        partial = self.env['message.wizard'].create({
+            'text': text
+        })
+        return {
+            'name': _("Message"),
+            'view_mode': 'form',
+            'view_id': False,
+            'view_type': 'form',
+            'res_model': 'message.wizard',
```

Comment:
message.wizard is from magento_bridge. You need to depend on that.

Reply:
Fixed.